client/my-sites/hosting/staging-site-card/index.js

```diff
@@ -227,14 +259,22 @@ const StagingSiteCard = ( { currentUserId, disabled, siteId, siteOwnerId, transl
 	);
 };
 
-export default connect( ( state ) => {
-	const currentUserId = getCurrentUserId( state );
-	const siteId = getSelectedSiteId( state );
-	const siteOwnerId = getSelectedSite( state )?.site_owner;
+export default withMediaStorage(
+	connect( ( state, { mediaStorage } ) => {
+		const currentUserId = getCurrentUserId( state );
+		const siteId = getSelectedSiteId( state );
+		const siteOwnerId = getSelectedSite( state )?.site_owner;
 
-	return {
-		currentUserId,
-		siteId,
-		siteOwnerId,
-	};
-} )( localize( StagingSiteCard ) );
+		let spaceQuotaExceededForStaging = false;
+		if ( mediaStorage?.storage_used_bytes ) {
+			spaceQuotaExceededForStaging =
+				mediaStorage.storage_used_bytes / GB_IN_BYTES > STAGING_CREATION_QUOTA_THRESHOLD;
```

Comment:
Here we assume that 100GB is half of the quota. However, the allowed quota can be different based on the plan or additional paid plugins or subscriptions. We could consider using `max_storage_bytes` here.

Reply:
I wasn't aware of that. I followed what the ticket was mentioned:
- If the site already exceeds 100 GB utilization, prevent staging sites from being created by disabling the button and showing a message.

That is a better solution though. I'll do that change!